Simplified/MyBooks/NYPLMyBooksDownloadCenter.m

```diff
@@ -1272,6 +1287,9 @@ - (void)cancelDownloadForBookIdentifier:(NSString *)identifier
     [[NYPLBookRegistry sharedRegistry]
      setState:NYPLBookStateDownloadNeeded forIdentifier:identifier];
   }
+#if FEATURE_AUDIOBOOKS
+  [self.audiobookDownloader cancelDownloadFor:identifier];
```

Comment:
are we positive `identifier` is always not nil? `cancelDownloadFor` accepts a non-optional.

Reply:
IIRC, a `nil` NSString will become an empty string when it reaches the Swift function. In our case, it will do nothing since none of the download objects will match with an empty book identifier.

Reply:
hmm, unless something changed recently, if `identifier`  is nil and it's passed to a swift function that accepts only a non-optional, that's a runtime error.... I believe I witnessed this kind of crash many times. Pls let me know if you saw something different. 

However, we should probably just add a guard at the beginning of `cancelDownloadForBookIdentifier:`, since if `identifier` is nil the rest of the method probably should (or _must_) be skipped entirely.